src/parser.ts

```diff
@@ -113,13 +117,14 @@ export class Parser extends DiagnosticEmitter {
   parseFile(
     text: string,
     path: string,
-    isEntry: bool
+    isEntry: bool,
+    sysPath?: string
```

Comment:
Is there maybe a way to map path to sysPath in asc itself? Seems unnecessary to pipe this through the compiler lib only to return it again.

Reply:
The parser needs to know about it because while parsing the file it associates the path in the system with imports in that file.  `asc` only knows about the backlog and uses the `getDependee` API to learn which file imported the file popped from the backlog.

Reply:
> it associates the path in the system with imports in that file

Hmm, so, it isn't possible, whenever an import becomes the next file, to obtain its dependee, and look up the sysPath for the respective dependee from a map kept by asc? Like, when asc makes a map of `file => sysPath` on adding to the backlog, then looking up either next's sysPath or its dependee's sysPath using that map?

Reply:
From asc's perspective it asks from the backlog the next file. And when it successful finds it calls out to parse it. So to do what you suggest it would need a copy of the back log to see the changes to it between parsing and asking for the next file.   So unless you want to add and API to get access to the backlog and not just the next file I don't see a way to do it.

Reply:
So, this isn't possible?

1. Ask for next file, locate it, add a mapping `map[nextFile] = sysPath`, parse it
2. Parsing adds additional files to backlog
3. Ask for next file from backlog, look up original `sysPath = map[getDependee(nextFile)]`

My understanding is that this way it is possible to recover the `sysPath` of the importing file, which would otherwise be passed around in compiler lib. From your comments it looks like I'm missing something, but I can't yet see what it is.

Reply:
Okay so now I changed the parser to track internal paths dependees and then use a external map to map those to sysPaths.